File: packages/urdu-js/tests/normalization/presentation.test.ts
import { describe, it, expect } from 'vitest'
import { normalizePresentationForms } from '../../src/normalization/presentation.js'

describe('normalizePresentationForms', () => {
  it('empty string returns empty', () => {
    expect(normalizePresentationForms('')).toBe('')
  })
  it('plain Urdu text passes through unchanged', () => {
    expect(normalizePresentationForms('پاکستان')).toBe('پاکستان')
  })
  it('presentation form ﻻ (U+FEFB) → لا', () => {
    // ARABIC LIGATURE LAM WITH ALEF ISOLATED FORM → lam + alef
    expect(normalizePresentationForms('\ufefb')).toBe('\u0644\u0627')
  })
  it('presentation form ﺎ (U+FE8E) → ا', () => {
    // ARABIC LETTER ALEF FINAL FORM → alef
    expect(normalizePresentationForms('\ufe8e')).toBe('\u0627')
  })
  it('mixed text: presentation form in sentence is normalized', () => {
    const result = normalizePresentationForms('کتاب\ufe8eہے')
    expect(result).toContain('\u0627')
    expect(result).not.toContain('\ufe8e')
  })
  it('characters outside presentation range pass through', () => {
    expect(normalizePresentationForms('علم')).toBe('علم')
  })
})

File: packages/urdu-dart/test/normalization/presentation_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('normalizePresentationForms', () {
    test('empty string returns empty',
        () => expect(normalizePresentationForms(''), equals('')));
    test('plain Urdu text passes through unchanged',
        () => expect(normalizePresentationForms('پاکستان'), equals('پاکستان')));
    test(
        'presentation form ﻻ (U+FEFB) -> لا',
        () => expect(
            normalizePresentationForms('\ufefb'), equals('\u0644\u0627')));
    test('presentation form ﺎ (U+FE8E) -> ا',
        () => expect(normalizePresentationForms('\ufe8e'), equals('\u0627')));
    test('mixed text: presentation form in sentence is normalized', () {
      final result = normalizePresentationForms('کتاب\ufe8eمیں');
      expect(result, contains('\u0627'));
      expect(result, isNot(contains('\ufe8e')));
    });
    test('characters outside presentation range pass through',
        () => expect(normalizePresentationForms('رحم'), equals('رحم')));
  });
}
